### PR TITLE
Makes antimatter code less shit

### PR DESCRIPTION
**Pull request by ghost**
:cl: Thefastfoodguy
fix: Antimatter shielding that can't find a control unit won't delete itself anymore
/:cl:

### CODE REVIEW

File: code/modules/power/antimatter/shielding.dm [hunk 1]
```diff
@@ -209,6 +210,7 @@
 	if(injecting_fuel && control_unit)
 		control_unit.exploding = 1
 	if(src)
+		visible_message("<span class='danger'>The [name] melts!</span>")
```
Comment (Cyberboss) [inline code/modules/power/antimatter/shielding.dm]: The [name] => [src]

File: code/modules/power/antimatter/shielding.dm [hunk 2]
```diff
@@ -35,7 +35,7 @@
 
 /obj/machinery/am_shielding/proc/controllerscan(priorscan = 0)
 	//Make sure we are the only one here
-	if(!istype(src.loc, /turf))
+	if(!istype(loc, /turf))
```
Comment (Cyberboss) [inline code/modules/power/antimatter/shielding.dm]: isturf(loc)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Supermichael777** [comment]
badminery detected

**Comment by MoreRobustThanYou** [comment]
the clown blew up cargo, not me

**Comment by Supermichael777** [comment]
if the controller dies presumably the thing should *checks physics textbook* Explode violently.

**Comment by Bawhoppen** [comment]
@Supermichael777 One change at a time. You don't always need to overhaul the entire feature in one PR.

**Review by AnturK — APPROVED** [review]

**Pull request merged**